### DISCUSSION TITLE
Connect to RabbitMQ from Java using Apache Qpid library

### DISCUSSION THREAD

**Discussion by bjek-123** [cloudamqp/amqproxy#163] [Q&A]
Hi, the AMQProxy looks exactly what we are after. We are connecting to an external broker on the internet that runs RabbitMQ. Our Java application resides in a secure network zone that aren't allowed to connect directly to the broker on the internet. We are using Apache Qpid client library to connect to the broker using AMQPS.
We just installed the amqproxy as a service on a Debian server and it runs fine on port 5673. The upstream url is set to amqps://tb.test.nhn.no. (default port)
When I run our Java program locally I get the error message "Failed to connect to remote at: amqps://some.server.net:5673" 
 We create a JMS Connection Factory with a few parameters, like username and password that the upstream broker wants but we use the local url to amqproxy. Should this work or are we trying to do something impossible here?

**Comment by bjek-123** [comment]
I can add that I managed to get an error message back now saying: 2024-05-13T10:44:29.233897Z DEBUG amq_proxy.server Client connection failure (192.168.160.119:62356) #<AMQProxy::Client::NegotiationError:Client negotiation failed>

**Reply by carlhoerberg** [reply]
Amqproxy doenst support TLS for client connections (if deployed locally it or in a secure network it should not be needed), so use amqp://some.server.net:5673 to connect to. Also make sure that you connect with the correct credentials, as those are passed through the proxy. That said we havent tried with the qpid client. We can do that, or if you're able to tcpdump 5672/5673 that's very helpful.

**Reply by carlhoerberg** [reply]
oh btw, `amqproxy` only speak AMQP 0-9-1, not 1.0. Which protocol do you try to connect with?

**Reply by carlhoerberg** [reply]
Looks like there are QPid JMS versions that support 0-9-1: https://qpid.apache.org/releases/qpid-jms-amqp-0-x-6.4.0/jms-amqp-0-8-book/index.html

**Comment by bjek-123** [comment]
Hej Carl, thanks for quick and thorough response 😊
Our broker (which is external and we don't have control over it) use AMQP 1.0. So I guess we can't use amqproxy.
Kind regards
Björn Eklund

Skickat från Outlook för iOS<https://aka.ms/o0ukef>
________________________________
Från: Carl Hörberg ***@***.***>
Skickat: Monday, May 13, 2024 6:43:57 PM
Till: cloudamqp/amqproxy ***@***.***>
Kopia: Björn Eklund ***@***.***>; Author ***@***.***>
Ämne: Re: [cloudamqp/amqproxy] Connect to RabbitMQ from Java using Apache Qpid library (Discussion #163)


Looks like there are QPid JMS versions that support 0-9-1: https://qpid.apache.org/releases/qpid-jms-amqp-0-x-6.4.0/jms-amqp-0-8-book/index.html<https://urldefense.proofpoint.com/v2/url?u=https-3A__qpid.apache.org_releases_qpid-2Djms-2Damqp-2D0-2Dx-2D6.4.0_jms-2Damqp-2D0-2D8-2Dbook_index.html&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=GD3_oT6TfOIiWhtuYckjdMSYhjUIcxSQrGbS3k57fqQ&m=tXSfPr7VrjY2vARjU9UFV3K9T6tVUPYMOQgfpyd0-fuI0DImUX3hge2ZlgUuPJeV&s=-VZIVg20XK551TX0fxrQLZX-4Q7H5Vt1V3yeUShbkPM&e=>

**Reply by carlhoerberg** [reply]
If it is a rabbitmq server it also speaks 0-9-1

**Reply by bjek-123** [reply]
Unfortunately it seems as if they only support amqp 1.0, [https://helsenorge.atlassian.net/wiki/spaces/HELSENORGE/pages/1656651781/AMQP+Profil]

They write that "AMQP versjon 1.0 ([OASIS Advanced Message Queuing Protocol (AMQP) Version 1.0](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-overview-v1.0-os.html)) skal benyttes i henhold til profilen under....."

**Reply by carlhoerberg** [reply]
Ok, might add amqp 1.0 support in the future.
